### DISCUSSION TITLE
GitHub Blocking AWS IP Addresses

### DISCUSSION THREAD

**Discussion by fletc3her** [community/community#23780] [New to GitHub]
<p>GitHub is blocking me from accessing either HTTPS or SSL endpoints from the 54.213.<em>.</em> IP address block which is used by EC2 at AWS. This is preventing my production and development machines which are virtually hosted from accessing my source code.</p>
<pre><code class="lang-auto"># time git pull
ssh: connect to host github.com port 22: Connection timed out
fatal: The remote end hung up unexpectedly

real 1m3.685s
user 0m0.028s
sys 0m0.079s
</code></pre>
<p>Testing SSL and HTTPS connectivity both time out from these servers but work from my local machines and from other IP address blocks at AWS.</p>
<pre><code class="lang-auto"># ssh -T git@github.com
# curl https://github.com/ -I
</code></pre>

**Comment by fletc3her** [comment]
<p>I have contacted support as well.</p>

**Comment by fletc3her** [comment, marked as answer]
<p>For some reason <a href="http://github.com" rel="nofollow noopener">github.com</a> was hard-coded in /etc/hosts.</p>
<p>I think that this was an old practice to prevent the system from adding many hosts to the known hosts file and prompting for permission every time.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Dropping the link to the <a href="https://github.community/t/cant-connect-from-aws/117221">other request here</a> as it might help someone else. Thanks for updating this post <a class="mention" href="/fletc3her">@fletc3her</a>!</p>

**Comment by mattnelsonuk** [comment]
I was also facing the same issue, thatnks to the comments above I traced my issue. 

I had a section in my ~/.ssh/config for `github.com` which was specifying the `host` separately as `ssh.github.com`. I think this fixed an issue in the past, but was no longer needed.
